go/ethadapter/obscuro_rpc_client.go

```diff
@@ -31,11 +31,17 @@ type obscuroWalletRPCClient struct {
 
 // NewObscuroRPCClient creates an obscuro RPC client for a given wallet, it will create and register a viewing key for the wallet as part of this setup
 func NewObscuroRPCClient(ipaddress string, port uint, wallet wallet.Wallet) (EthClient, error) {
-	client, err := rpcclientlib.NewViewingKeyNetworkClient(fmt.Sprintf("%s:%d", ipaddress, port))
+	vk, publicVK, signedVK, err := viewkey.GenerateAndSignViewingKey(wallet)
 	if err != nil {
 		return nil, err
 	}
-	err = viewkey.GenerateAndRegisterViewingKey(client, wallet)
+	rpcURL := fmt.Sprintf("%s:%d", ipaddress, port)
+	client, err := rpcclientlib.NewViewingKeyNetworkClient(rpcURL, wallet, vk, publicVK, signedVK)
+	if err != nil {
+		return nil, err
+	}
+
+	err = client.RegisterViewingKeyWithEnclave(publicVK, signedVK)
```

Comment:
@BedrockSquirrel Shouldn't you do this directly in the constructor?

Reply:
Yeah I suppose so, I guess I was worried it's a bit of a magic side-effect to have a constructor make an RPC call but constructors are a weird concept in Go aren't they. Will change it (removes a duplication too I think).

I think it registering the ViewingKey is what Tudor has in mind because he suggested the AuthClient constructor we end up with takes the VK and the signed VK so makes sense that it then registers itself at that 'construction' time.

Reply:
Oh, it was already registered in the NewViewingKeyClient, this was registering it twice.